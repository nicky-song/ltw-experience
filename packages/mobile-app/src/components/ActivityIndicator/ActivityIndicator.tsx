import {
  ActivityIndicator as ActivityIndicatorRN,
  ActivityIndicatorProps,
} from 'react-native';
export function ActivityIndicator(
  props: Omit<ActivityIndicatorProps, 'color'>,
) {
  return (
    <ActivityIndicatorRN
      size='large'
      {...props}
      color='#6FC07A'
      testID={'activity-indicator'}
    />
  );
}
