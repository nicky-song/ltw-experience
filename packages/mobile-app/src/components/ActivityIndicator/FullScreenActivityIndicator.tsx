import { StyleSheet, View } from 'react-native';
import { Text } from '@ant-design/react-native';
import { ActivityIndicator } from './ActivityIndicator';

export const FullScreenActivityIndicator = ({ text }: { text?: string }) => (
  <View style={styles.container}>
    {!!text && <Text style={styles.loadingText}>{text}</Text>}
    <ActivityIndicator />
  </View>
);

const styles = StyleSheet.create({
  container: {
    paddingTop: 150,
    height: '100%',
  },
  loadingText: {
    fontSize: 16,
    fontWeight: '400',
    marginBottom: 40,
    textAlign: 'center',
  },
});
