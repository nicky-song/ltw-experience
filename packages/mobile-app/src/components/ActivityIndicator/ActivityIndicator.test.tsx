import { ActivityIndicator } from './ActivityIndicator';
import renderer from 'react-test-renderer';

describe('ActivityIndicator', () => {
  it('should render', () => {
    const activityIndicator = renderer.create(<ActivityIndicator />);
    expect(activityIndicator).toMatchSnapshot();
  });

  it('renders Small', () => {
    const activityIndicator = renderer.create(
      <ActivityIndicator size='small' />,
    );
    expect(activityIndicator).toMatchSnapshot();
  });
});
