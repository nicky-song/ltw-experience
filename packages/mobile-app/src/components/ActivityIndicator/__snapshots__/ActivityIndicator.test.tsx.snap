// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityIndicator renders Small 1`] = `
<ActivityIndicator
  color="#6FC07A"
  size="small"
  testID="activity-indicator"
/>
`;

exports[`ActivityIndicator should render 1`] = `
<ActivityIndicator
  color="#6FC07A"
  size="large"
  testID="activity-indicator"
/>
`;
